[docs] Quicktour (#2456)
huggingface/accelerate: docs/source/basic_tutorials/migration.md

  for input, label in training_dataloader:
+     with accelerator.accumulate(model):
          predictions = model(input)
          loss = loss_function(predictions, label)
          accelerator.backward(loss)
          optimizer.step()
          scheduler.step()
          optimizer.zero_grad()
```

### Gradient clipping

Gradient clipping is a technique to prevent "exploding gradients", and Accelerate offers:

* [`~Accelerator.clip_grad_value_`] to clip gradients to a minimum and maximum value
* [`~Accelerator.clip_grad_norm_`] for normalizing gradients to a certain value

### Mixed precision

Mixed precision accelerates training by using a lower precision data type like fp16 (half-precision) to calculate the gradients. For the best performance with Accelerate, the loss should be computed inside your model (like in Transformers models) because computations outside of the model are computed in full precision.

Set the mixed precision type to use in the [`Accelerator`], and then use the [`~Accelerator.autocast`] context manager to automatically cast the values to the specified data type.

```diff
+ accelerator = Accelerator(mixed_precision="fp16")
+ with accelerator.autocast():

>       loss = complex_loss_function(outputs, target):

Reviewer: We should be careful here and mention that this (`autocast`) is only needed if there is some *other* thing in mixed precision that needs to be done outside of just what happens during `.backward()` on the loss, as accelerate already does the scaling during `.backward()`. 

As automatic mixed precision is one of our advertisements :)

```

With mixed precision training, it may skip a few gradient updates at the beginning and during training because of the dynamic loss scaling strategy. This strategy reduces the loss scaling factor when gradients overflow during training to avoid it happening again at the next step. This means the learning rate scheduler may be updated even though there was no update, which can impact the training process when you have very little training data or if the first learning rate values of your scheduler are very important. You can prevent the learning rate scheduler from updating if the optimizer wasn't updated with the [`~Accelerator.optimizer_step_was_skipped`] method.

```py
if not accelerator.optimizer_step_was_skipped:
    lr_scheduler.step()

> ```

Reviewer: I'm not sure this should be here at all, as we handle this internally and can be a bit confusing for users? (as you have to go against the grain and set `step_scheduler_with_optimizer=False` to even reach this point)

Author: Ah ok, I'll take it out then to avoid confusion!

## Save and load

Accelerate can also save and load a *model* once training is complete or you can also save the model and optimizer *state* which could be useful for resuming training.

### Model

> Once all processes are complete, unwrap the model with the [`~Accelerator.unwrap_model`] method before saving it because the [`~Accelerator.prepare`] method may have placed your model inside a larger model for distributed training. If you don't unwrap the model, saving the model state dictionary also saves any potential extra layers from the larger model and you won't be able to load the weights back into your base model.

Reviewer: ```suggestion
Once all processes are complete, unwrap the model with the [`~Accelerator.unwrap_model`] method before saving it because the [`~Accelerator.prepare`] method will have wrapped your model into the proper interface for distributed training. If you don't unwrap the model, saving the model state dictionary also saves any potential extra layers from the larger model and you won't be able to load the weights back into your base model.
```
